index.js

```diff
@@ -0,0 +1,56 @@
+const defaults = {
+  routeRules: [
+    '.php',
+    '.aspx',
+    '.ashx',
+    '.ico',
+    '/feed',
+    '/rss',
+    '/null',
+    '/undefined',
+    '/...',
+    '/true',
+    '/false'
+  ],
+  userAgentRules: [
+    'mj12bot' // Generates excessive 404s
+  ]
+};
+
+const register = function(server, options) {
+  const settings = Object.assign({}, defaults, options);
+
+  server.ext('onPreResponse', (request, h) => {
+    const userAgent = request.headers['user-agent'].toLowerCase();
+    const acceptsHtml = (request.headers.accept || '').includes('html');
+    let matched = false;
+
+    if (acceptsHtml) {
+      return h.continue;
```

Comment:
Is this going to break any API routes (in a general sense)... Or is this lib intended to only go on say "web" packages?

Reply:
It shouldn't unless the API tries to hit one of the routes in the rules or sets a bad UA.